go/integration_test/processor/processor_gas_test.go

```diff
@@ -0,0 +1,503 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package processor
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/processor/floria"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	op "github.com/ethereum/go-ethereum/core/vm"
+)
+
+func getGasTestScenarios() map[string]Scenario {
+	const (
+		codePrice = 3 + 3
+		excessGas = 100
+	)
+
+	allTestCases := map[string]Scenario{
+		"ValueTransferExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas,
```

Comment:
nit: it is a bit odd to use constants from a concrete implementation to define implementation agnostic tests. It might be nice to have those constants defined at some implementation-independent place, but I am not sure what a good place might be. So I guess we leave this for a potential future cleanup pass ...

Reply:
I like the idea of implementation agnostic tests, I would say that if we want these values to also be called in the implementations, then a nice place would be on the same level as the processor interface definition. But if we just want these values to be test constants, then they could be defined in `go/integration_tests/processor/gas_values.go` . wdyt ?

Reply:
I will keep this as a cleanup task once the implementation is fixed. I think all these constants should be in the same place eventually.